src/modules/assessments/components/household/HouseholdAssessmentTabPanel.tsx

```diff
@@ -98,95 +96,180 @@ const HouseholdAssessmentTabPanel = memo(
-          buttonProps: {
-            disabled: !previousTab,
-            startIcon: <ArrowBackIcon fontSize='small' />,
-            ...nextPreviousProps,
-          } as const,
-          onSuccess: () => {
-            if (previousTab) navigateToTab(previousTab);
-          },
-        });
+    const onCompletedMutation = useCallback(
+      (status: AssessmentResponseStatus) => {
+        // console.debug('Completed with status', status);
+        if (['saved', 'submitted'].includes(status)) {
+          onFormStateChange?.(enrollmentId, 'saveCompleted');
+        } else if (['warning', 'error'].includes(status)) {
+          // Treat warning and errors the same right now. This should be expanded to treat them differently with visual indicators.
+          onFormStateChange?.(enrollmentId, 'saveFailed');
+        }
 
-        if (assessment && !assessment.inProgress) {
-          config.push({
-            id: 'submit',
-            label: 'Save & Submit',
-            centerAlign: true,
-            action: FormActionTypes.Submit,
-            buttonProps: { variant: 'outlined' } as const,
-            onSuccess: () => {
-              updateTabStatus(AssessmentStatus.Submitted, id);
-              refetch();
-            },
-          });
-        } else {
-          config.push({
-            id: 'save',
-            label: 'Save Assessment',
-            centerAlign: true,
-            action: FormActionTypes.Save,
-            buttonProps: { variant: 'contained', size: 'large' } as const,
-            onSuccess: () => {
-              if (!assessment) updateTabStatus(AssessmentStatus.Started, id);
-              refetch();
-            },
-          });
+        // If this was a brand new assessment being saved for the first time,
+        // we need to refetch to get it.
+        if (status === 'saved' && !assessmentId) {
+          updateTabStatus(AssessmentStatus.Started, id);
+          refetch();
         }
+      },
+      [
+        assessmentId,
+        enrollmentId,
+        id,
+        onFormStateChange,
+        refetch,
+        updateTabStatus,
+      ]
+    );
 
-        config.push({
-          id: 'next',
-          label: 'Next',
-          action: FormActionTypes.Navigate,
-          buttonProps: {
-            disabled: !nextTab,
-            endIcon: <ArrowForwardIcon fontSize='small' />,
-            ...nextPreviousProps,
-          } as const,
-          onSuccess: () => {
-            if (nextTab) navigateToTab(nextTab);
+    const { submitHandler, saveDraftHandler, mutationLoading, errors } =
+      useAssessmentHandlers({
+        definition,
+        enrollmentId,
+        assessmentId,
+        assessmentLockVersion: assessment?.lockVersion,
+        onCompletedMutation,
+      });
+
+    // disabling nav buttons while loading, and if there are errors
+    const disableNavigation = useMemo(
+      () => mutationLoading || formState.errors,
+      [formState.errors, mutationLoading]
+    );
```

Comment:
micro optimization, maybe skip the memoization for simple calculations that return a primitive (so same object identity)
```suggestion
    const disableNavigation = mutationLoading || formState.errors 
```